ocs_ci/deployment/factory.py

```diff
@@ -30,6 +31,7 @@ def __init__(self):
             "baremetal_upi": BAREMETALUPI,
             "gcp_ipi": GCPIPI,
             "powervs_upi": IBMDeployment,
+            "ibm_cloud_managed": IBMCloud,
```

Comment:
I see that we have all platforms classes was just curious what is the general purpose of `factory.py` :)

Reply:
It's one of the common design pattern of object oriented programming ;) you can try to google it ;) https://www.tutorialspoint.com/design_pattern/factory_pattern.htm this is one of the first result.